Mutex.php

```diff
@@ -10,204 +10,149 @@
+     * @param string $name of the lock to be released. This lock must already exist.
+     * @return bool lock release result: `false` in case named lock was not found or Redis command failed.
      */
     protected function releaseLock($name)
     {
-        list($lockKey, $lockValue) = $this->getLock($name);
-
-        if (null === $lockKey) {
+        static $releaseLuaScript = <<<LUA
+if redis.call("GET",KEYS[1])==ARGV[1] then
+    return redis.call("DEL",KEYS[1])
+else
+    return 0
+end
+LUA;
+        if (!isset($this->_lockValues[$name]) || !$this->redis->executeCommand('EVAL', [
+                $releaseLuaScript,
+                1,
+                $this->calculateKey($name),
+                $this->_lockValues[$name]
+            ])) {
             return false;
-        }
-
-        $luaScript = 'if redis.call("GET", KEYS[1]) == ARGV[1] then
-                        return redis.call("DEL", KEYS[1])
-                    else
-                        return 0
-                    end';
-
-        if ($this->redis->executeCommand('EVAL', [$luaScript, 1, $lockKey, $lockValue])) {
-            $this->deleteLock($name);
-
+        } else {
+            unset($this->_lockValues[$name]);
             return true;
         }
-
-        return false;
-    }
-
-    /**
-     * Generates redis key for a lock name.
-     *
-     * @param string $lockName
-     *
-     * @return string
-     */
-    protected function getLockKey($lockName)
-    {
-        return $this->keyPrefix . $lockName;
-    }
-
-    /**
-     * Gets lock from the local map
-     *
-     * @param string $name
-     *
-     * @return array|null lock's [keyName, keyValue] pair for redis or null if there is no such lock
-     */
-    protected function getLock($name)
-    {
-        return isset($this->_lockMap[$name]) ? [$this->getLockKey($name), $this->_lockMap[$name]] : null;
-    }
-
-    /**
-     * Adds lock to the local map
-     *
-     * @param string $name
-     *
-     * @return array|null lock's [keyName, keyValue] pair for redis or null if it's already exists in $lockMap
-     */
-    protected function addLock($name)
-    {
-        if (isset($this->_lockMap[$name])) {
-            return null;
-        }
-
-        $lockValue = Yii::$app->security->generateRandomString();
-        $this->_lockMap[$name] = $lockValue;
-
-        return [$this->getLockKey($name), $lockValue];
     }
 
     /**
-     * Deletes lock from the local map
-     *
-     * @param string $name
-     *
-     * @return bool
+     * Generates a unique key used for storing the mutex in Redis.
+     * @param string $name mutex name.
+     * @return string a safe cache key associated with the mutex name.
      */
-    protected function deleteLock($name)
+    protected function calculateKey($name)
     {
-        unset($this->_lockMap[$name]);
+        return $this->keyPrefix . md5(json_encode([__CLASS__, $name]));
```

Comment:
> If I want to implement a touch() method to prolong a mutex expire time in long code blocks, I need to know a key name.

Personally, I tend to see 'calculating' and 'getting' to be different types of logic of which the first is preferably private (you calculate once) and the getter protected (which can be reused). I am trying to grasp the usecase for having a calculating function to be overrided. After all, you can already set the $keyPrefix.